vendor/bat-native-ledger/src/bat/ledger/internal/core/bat_ledger_task.h

```diff
@@ -0,0 +1,59 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_VENDOR_BAT_NATIVE_LEDGER_SRC_BAT_LEDGER_INTERNAL_CORE_BAT_LEDGER_TASK_H_
+#define BRAVE_VENDOR_BAT_NATIVE_LEDGER_SRC_BAT_LEDGER_INTERNAL_CORE_BAT_LEDGER_TASK_H_
+
+#include <utility>
+
+#include "base/bind.h"
+#include "bat/ledger/internal/core/async_result.h"
+#include "bat/ledger/internal/core/bat_ledger_context.h"
+
+namespace ledger {
+
+// Convenience class for defining task components that can be started by calling
+// |BATLedgerContext::StartTask|. Subclasses must implement a |Start| method
+// that begins the asynchronous operation.
```

Comment:
Wasn't quite sure if we're enforcing this constraint with a pure virtual function or something similar?

Reply:
The constraint is enforced by usage of `BATLedgerContext::StartTask<T>`, which expects that `T` has:

- A `result()` accessor
- A `void Start(Args...)` method, where `Args...` can be any sequence of arguments that `StartTask<T>` will forward to the task.

Perhaps I could add some additional usage documentation.

Reply:
OK, perfect.